app/domain/authentication/authn_jwt/identity_providers/create_identity_provider.rb

```diff
@@ -0,0 +1,62 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Factory for jwt identity providers.
+      # If Identity variable is configured factory return the decoded_token_provider
+      # If the Identity variable is not configured and there is account field in url factory returns url provider
+      # If the above conditions are not met exception is raised
+      CreateIdentityProvider = CommandClass.new(
+        dependencies: {
+          identity_from_url_provider_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromUrlProvider,
+          identity_from_decoded_token_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromDecodedTokenProvider,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        def call
+          create_identity_provider
+        end
+
+        private
+
+        def identity_from_decoded_token_provider
+          @identity_from_decoded_token_provider ||= @identity_from_decoded_token_class.new(@authentication_parameters)
+        end
+
+        def identity_from_url_provider
+          @identity_from_url_provider ||= @identity_from_url_provider_class.new(@authentication_parameters)
+        end
+
+        def create_identity_provider
+          @logger.debug(LogMessages::Authentication::AuthnJwt::SelectingIdentityProviderInterface.new)
+
+          check_identity_configuration
+          if identity_from_decoded_token_provider.identity_available
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedIdentityProviderInterface.new(
+                TOKEN_IDENTITY_PROVIDER_INTERFACE_NAME
+              )
+            )
+            return identity_from_decoded_token_provider
+          elsif identity_from_url_provider.identity_available
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedIdentityProviderInterface.new(
+                URL_IDENTITY_PROVIDER_INTERFACE_NAME
+              )
+            )
+            return identity_from_url_provider
+          end
+        end
+
+        def check_identity_configuration
```

Comment:
i would change the name to `validate_identity_configuration` because you raise an error if the validation fails (and not just check)